### PR TITLE
expost sst reader

### PR DESCRIPTION
**Pull request by UncP**

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
Please add some tests for SstFileReader.

**Comment by UncP** [comment]
@zhangjinpeng1987 PTAL

**Comment by zhangjinpeng1987** [comment]
I means add unit tests for SstFileReader.

**Comment by UncP** [comment]
@zhangjinpeng1987 PTAL

**Comment by zhangjinpeng1987** [comment]
@huachaohuang PTAL

File: include/rocksdb/sst_file_reader.h [hunk 1]
```diff
@@ -5,22 +5,60 @@
...
 
 namespace rocksdb {
 
+struct ImmutableCFOptions;
+class RandomAccessFileReader;
+class InternalKeyComparator;
+class TableReader;
+struct TableBuilderOptions;
+
+struct DefaultKvHandler {
```
Comment (zhangjinpeng1987) [inline include/rocksdb/sst_file_reader.h]: I think these `DefaultXXHandler` should be moved out of this file.

Comment (UncP) [reply include/rocksdb/sst_file_reader.h]: Move it to another file?

**Comment by UncP** [comment]
@zhangjinpeng1987 @huachaohuang PTAL, make check passed.

**Comment by huachaohuang** [comment]
The `SstFileReader` seems like a class for the tool, why not just add a wrapper for `TableReader`?

**Comment by UncP** [comment]
What do you mean? There is already a `TableReader` in SstFileReader.
Also, since there is `SstFileWriter`, there should be a corresponding reader.
@huachaohuang

File: tools/Makefile [hunk 2]
```diff
@@ -0,0 +1,594 @@
+# CMAKE generated file: DO NOT EDIT!
```
Comment (zhangjinpeng1987) [inline tools/Makefile]: Why so many changes in this file?

File: table/sst_file_reader_test.cc [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    tb->Add(MakeKey(i), MakeValue(i));
+  }
+  tb->Finish();
+  file_writer->Close();
+}
+
+void cleanup(const std::string& file_name) {
+  Env* env = Env::Default();
+  env->DeleteFile(file_name);
+  std::string outfile_name = file_name.substr(0, file_name.length() - 4);
```
Comment (siddontang) [inline table/sst_file_reader_test.cc]: what is this?

File: tools/Makefile (hunk 2 above)
Comment (UncP) [reply tools/Makefile]: I deleted this file.

File: table/sst_file_reader_test.cc (hunk 3 above)
Comment (UncP) [reply table/sst_file_reader_test.cc]: to clean up generated sst file by test

**Comment by huachaohuang** [comment]
I mean, we don't even need to modify RocksDB. `TableReader` already provides all the information we need, we just need to create a wrapper for it.
/cc @siddontang @zhangjinpeng1987

**Comment by UncP** [comment]
Do you mean add a wrapper in rust-rocksdb? @huachaohuang

**Comment by UncP** [comment]
I read the code again, I think we still need `sst file reader`, `table reader` is not in `include` dir, we still need to change rocksdb code to use it, also, when we initialize `table reader`, we need some options, `sst file reader` helps us initialize those options, and if we add a wrapper in rust-rocksdb, it doesn't help simplify things.

**Pull request closed without merging**